core/forms.py

```diff
@@ -14,6 +21,87 @@ def clean(self, values):
         return super(BetterReCaptchaField, self).clean(values)
 
 
+class AddOrganizerForm(forms.Form):
+    """
+    Custom form for adding new organizers to an existing event. 
+    
+    If user of given email already exists, they're added to the event and 
+    receive e-mail notification about it. 
+    
+    If user is new, they're created (randomly generated password), invited
+    to Slack and receive e-mail notification with instructions to login 
+    (including password).
+    """
+    event = forms.ModelChoiceField(queryset=Event.objects.all())
+    name = forms.CharField(label="Organizer's full name")
+    email = forms.CharField(label="E-mail address")
+
+    def __init__(self, event_choices=None, *args, **kwargs):
+        super(AddOrganizerForm, self).__init__(*args, **kwargs)
+        if event_choices:
+            self.fields['event'].queryset = event_choices
+
+    def generate_password(self):
+        return ''.join(random.choice(string.ascii_lowercase + string.digits) for _ in range(8))
+
+    def invite_to_slack(self, email, name):
+        try:
+            user_invite(email, name)
+        except (ConnectionError, SlackerError) as e:
+            self._errors.append('Slack invite unsuccessful, reason: {}'.format(e))
+
+    def notify_existing_user(self, user):
+        content = render_to_string('emails/existing_user.html', {
+            'user': user,
+            'event': self.cleaned_data['event']
+        })
+        subject = 'You have been granted to new Django Girls event'
+        self.send_email(content, subject, user)
+
+    def notify_new_user(self, user):
+        content = render_to_string('emails/new_user.html', {
+            'user': user,
+            'event': self.cleaned_data['event'],
+            'password': self._password,
+            'errors': self._errors,
+        })
+        subject = 'Access to Django Girls website'
+        self.send_email(content, subject, user)
+
+    def send_email(self, content, subject, user):
+        send_mail(subject, content, "Django Girls <hello@djangogirls.org>",
+            [user.email], fail_silently=True)
+
+    def save(self, *args, **kwargs):
+        assert self.is_valid()
+        self._errors = []
+        email = self.cleaned_data['email']
+        event = self.cleaned_data['event']
+        if not User.objects.filter(email=email).exists():
```

Comment:
Ideally you could use `User.objects.get_or_create` to avoid multiple queries to the User table 😀